skyfield/data/mpc.py

```diff
@@ -226,6 +226,30 @@ def comet_orbit(row, ts, gm_km3_s2):
     comet._rotation = inertial_frames['ECLIPJ2000'].T
     return comet
 
+def _comet_orbits(rows, ts, gm_km3_s2):
+    e = rows.eccentricity.values
+    p = 1 - e**2
+    parabolic = (e == 1.0)
+    p[parabolic] = rows.perihelion_distance_au.values[parabolic] * 2.0
```

Comment:
Nice, I like how you improved this section! I think you can make a final improvement: move `* rows.perihelion_distance_au.values` into the initial formula for `p`. That should avoid having to index into the big array twice with two different Boolean masks, and building those two subarrays. Then these two lines, I think, simply become `*= 2.0` and `/= 1.0 - e[~parabolic])`?

Reply:
I tried this but unfortunately in the parabolic case p is always initialized as 0 because 1-e**2 is zero, making the perihelion distance disappear.

Another option is this:
```
    e = rows.eccentricity.values
    p = (1 - e*e) * rows.perihelion_distance_au.values/ (1.0 - e)
    p[e==1.0] = rows.perihelion_distance_au.values[e==1.0] * 2.0
```
which works because the values are `np.float64` and any division by zero in the second line is turned into `np.inf`, which would later be replaced in the 3rd line. The downside is that this generates a warning, which we should probably avoid so users don't think something is wrong.